### DISCUSSION TITLE
Renaming the default branch to `main`

### DISCUSSION THREAD

**Discussion by al-fisher** [BHoM/BHoM#1087] [General] [answered]
GitHub have now started the process of renaming default branches from `master` to `main`.

See https://github.com/github/renaming

Language is important, and thus proposing we in BHoM similarly adopt this positive change and move to using `main` for all our base branches, across all BHoM repositories, removing reference to `master`.

At the BHoM organisation level the default branch name has already now been updated to `main`.
_However, all the existing repos, including the template repo, have not yet been renamed._ 

GitHub are working to make transitions for existing repos a little easier - see [comments here](https://github.com/github/renaming#later-this-year) - so proposing we monitor and do not update our repos immediately.

As discussed with @FraserGreenroyd we will follow their advice and make the switch when some of the seamless transition and retargeting is in place.
We will revisit and hope to action as part of our 4.0 milestone close out and start of year administrative tasks.

So no action for all currently - but keep an eye out for updates

**Comment by al-fisher** [comment]
@FraserGreenroyd - looks like the advice and ease around updating default branch has been revised since this issue was raised.
https://github.com/github/renaming#renaming-existing-branches
Worth us relooking now and planning right time to action - actual branch renaming looks very simple now - but will be interested in thoughts on additional places and settings that might be affected in say CI/CD and @BHoMBot processes.

**Reply by FraserGreenroyd** [reply]
This is something I was planning to look at in the 5.0 milestone now they've rolled out the renaming functionality more generally. It will require an update to BHoMBot processes, and possibly some processes of @adecler around data migration processes that exist (tagging @adecler to confirm/deny that as appropriate), so not something being targeted in this beta package but for early-mid 5.0 :smile:

**Reply by al-fisher** [reply]
OK - perfect! Thanks @FraserGreenroyd

**Comment by FraserGreenroyd** [comment, marked as answer]
This has now been completed with the primary branch now becoming `main` and has been rolled out to all repositories within the BHoM organisation :smile:
